tests/model_selection/test_incremental.py

```diff
@@ -115,8 +118,15 @@ def _additional_calls(info):
     for key in keys:
         del models[key]
 
-    while c.futures or s.tasks:  # Make sure cleans up cleanly after running
+    # Make sure cleans up quickly after running
+    deadline = time() + 5
+    while c.futures:
```

Comment:
What tasks are hanging around?  Printing out `s.tasks` and seeing what is sticking around and why would probably help isolate the problem here pretty quickly.

Reply:
It looks like calls to `partial_fit` are staying in state "memory" instead of moving to state "released." I couldn't find the reason why in my poking around. Here's some more context for one specific task:

<details>

Here's some of `s.tasks` when the `ValueError` is raised:

``` 
{"('array-5c6c0aea4775b68f8400b686cdf361ee', 0)": <Task "('array-5c6c0aea4775b68f8400b686cdf361ee', 0)" released>,
 "('astype-getitem-a0e460089e0b0c5d0baf1ca408aaff5b', 0)": <Task "('astype-getitem-a0e460089e0b0c5d0baf1ca408aaff5b', 0)" released>,
 "('astype-getitem-eb01c7dcea1f24ea913fd665b30d03a4', 0)": <Task "('astype-getitem-eb01c7dcea1f24ea913fd665b30d03a4', 0)" released>,
 "('getitem-5342f25119d203e5d67f88228f1e00b5', 0, 0)": <Task "('getitem-5342f25119d203e5d67f88228f1e00b5', 0, 0)" released>,
 "('getitem-99834584206c72f1f1a79495c4c62593', 0, 0)": <Task "('getitem-99834584206c72f1f1a79495c4c62593', 0, 0)" released>,
 "('getitem-a0e460089e0b0c5d0baf1ca408aaff5b', 0)": <Task "('getitem-a0e460089e0b0c5d0baf1ca408aaff5b', 0)" released>,
 "('getitem-c0cd079c7dd295b1df3a8ceb8158f96c', 0, 0)": <Task "('getitem-c0cd079c7dd295b1df3a8ceb8158f96c', 0, 0)" released>,
 "('getitem-c0cd079c7dd295b1df3a8ceb8158f96c', 1, 0)": <Task "('getitem-c0cd079c7dd295b1df3a8ceb8158f96c', 1, 0)" released>,
 "('getitem-eb01c7dcea1f24ea913fd665b30d03a4', 0)": <Task "('getitem-eb01c7dcea1f24ea913fd665b30d03a4', 0)" released>,
 "('normal-71ffc5749897310d732642de69ab3045', 0, 0)": <Task "('normal-71ffc5749897310d732642de69ab3045', 0, 0)" released>,
 "('normal-71ffc5749897310d732642de69ab3045', 1, 0)": <Task "('normal-71ffc5749897310d732642de69ab3045', 1, 0)" released>,
 "('normal-getitem-5342f25119d203e5d67f88228f1e00b5', 0, 0)": <Task "('normal-getitem-5342f25119d203e5d67f88228f1e00b5', 0, 0)" released>,
 "('normal-getitem-c0cd079c7dd295b1df3a8ceb8158f96c', 0, 0)": <Task "('normal-getitem-c0cd079c7dd295b1df3a8ceb8158f96c', 0, 0)" released>,
 "('random_sample-d4e3c66fd779378c4e00049a86312ef1', 0)": <Task "('random_sample-d4e3c66fd779378c4e00049a86312ef1', 0)" released>,
 "('random_sample-d4e3c66fd779378c4e00049a86312ef1', 1)": <Task "('random_sample-d4e3c66fd779378c4e00049a86312ef1', 1)" released>,
 "('sum-aggregate-eb653519b96897acbc044f0da8455aec', 0)": <Task "('sum-aggregate-eb653519b96897acbc044f0da8455aec', 0)" released>,
 "('sum-aggregate-eb653519b96897acbc044f0da8455aec', 1)": <Task "('sum-aggregate-eb653519b96897acbc044f0da8455aec', 1)" released>,
 "('sum-sum-aggregate-eb653519b96897acbc044f0da8455aec', 0)": <Task "('sum-sum-aggregate-eb653519b96897acbc044f0da8455aec', 0)" released>,
 "('sum-sum-aggregate-eb653519b96897acbc044f0da8455aec', 1)": <Task "('sum-sum-aggregate-eb653519b96897acbc044f0da8455aec', 1)" released>,
 '_create_model-204466dbc69f6ce91aae7a8f90e696f8': <Task '_create_model-204466dbc69f6ce91aae7a8f90e696f8' released>,
 '_create_model-3499389710d7a55287c8e1fd9d3e8a86': <Task '_create_model-3499389710d7a55287c8e1fd9d3e8a86' released>,
 '_partial_fit-20faea901cc3feebc39428be644dce28': <Task '_partial_fit-20faea901cc3feebc39428be644dce28' released>,
 '_partial_fit-36af93d60c2f4fcd5e15cbbceef5b79f': <Task '_partial_fit-36af93d60c2f4fcd5e15cbbceef5b79f' memory>,
 '_partial_fit-593a8e400151a3dea752eab563d2c641': <Task '_partial_fit-593a8e400151a3dea752eab563d2c641' memory>,
 '_partial_fit-5c386d4ccf33082fcec2b1c3aecb9e9a': <Task '_partial_fit-5c386d4ccf33082fcec2b1c3aecb9e9a' released>,
 '_partial_fit-6ab6fa94ce10d5e2977cb93ec6b585c3': <Task '_partial_fit-6ab6fa94ce10d5e2977cb93ec6b585c3' released>,
 '_partial_fit-842952cf6460cfad15896c7e8d7a673d': <Task '_partial_fit-842952cf6460cfad15896c7e8d7a673d' memory>,
 '_partial_fit-95daf36e4defa195bab22b49f709efd7': <Task '_partial_fit-95daf36e4defa195bab22b49f709efd7' released>,
 '_partial_fit-99f97f6b969121825fd2b90aad30381c': <Task '_partial_fit-99f97f6b969121825fd2b90aad30381c' released>,
 '_partial_fit-a20889336cd81797af3da7d22321b34f': <Task '_partial_fit-a20889336cd81797af3da7d22321b34f' memory>,
 '_partial_fit-ab249a7357580aa18fe2eba60be72bcb': <Task '_partial_fit-ab249a7357580aa18fe2eba60be72bcb' released>,
 '_score-15cae06b53fe052a9f7ffbc524357d5f': <Task '_score-15cae06b53fe052a9f7ffbc524357d5f' memory>,
 '_score-b79d83e626689a3b49e0dbf115a80ec3': <Task '_score-b79d83e626689a3b49e0dbf115a80ec3' memory>,
 'finalize-285f1af18fd76cac82cce5d7778d8044': <Task 'finalize-285f1af18fd76cac82cce5d7778d8044' released>,
 'finalize-e5aafd4559bab198c2561658595bfd3e': <Task 'finalize-e5aafd4559bab198c2561658595bfd3e' released>}
```

If I assign `key = "_partial_fit-593a8e400151a3dea752eab563d2c641"`, then

``` python
>>> task = s.tasks[key]
>>> task
<Task '_partial_fit-593a8e400151a3dea752eab563d2c641' memory>
>>> task.who_has
{<Worker 'tcp://127.0.0.1:60747', name: 1, memory: 4, processing: 0>}
>>> task.waiters
set()
>>> task.dependents
set()
>>> task.dependencies
{<Task '_partial_fit-a20889336cd81797af3da7d22321b34f' memory>,
 <Task "('getitem-99834584206c72f1f1a79495c4c62593', 0, 0)" released>,
 <Task "('getitem-eb01c7dcea1f24ea913fd665b30d03a4', 0)" released>}
```

</details>

Reply:
You might take a look at `s.clients` to view the Scheduler's perspective of all of the clients and what they want.

1.  Is there a phantom client lying around maybe holding onto a future stopping it from being released?
2.  Does the scheduler's perspective of `ClientState.wants_what` match the state of `Client.futures`?  If not, what is misaligned?  What are the stories of those tasks (see the `s.story` method.

I'm *hoping* that this is interesting/educational for you.  It seems like a nice opportunity to dive into the scheduler a bit.  If you're very frustrated at this point let me know and I can dive in instead.

Reply:
If it's useful I'd also be happy to jump on a 30 minute screenshare call.  That might make debugging this sort of thing easier.

Reply:
Thanks for checking in @mrocklin. I've been busy improving [scikeras][1] this week; I'd be happy to hop on a call after I spend a little longer on this PR.

[1]:https://github.com/adriangb/scikeras